opencog/ure/forwardchainer/ForwardChainer.cc

```diff
@@ -411,6 +412,28 @@ HandleSet ForwardChainer::apply_rule(const Rule& rule)
 		AtomSpace derived_rule_as(&ref_as);
 		Handle rhcpy = derived_rule_as.add_atom(rule.get_rule());
 
+		HandleSeq clauses = rule.get_clauses();
+		Handle vdecl = rule.get_vardecl();
+		HandleSet varset1;
+		HandleSet varset2;
+
+		if (vdecl != Handle::UNDEFINED)
+		{
+			if (VariableListCast(vdecl))
+				varset1 = VariableListCast(vdecl)->get_variables().varset;
+			else
+				varset1.insert(vdecl);
+
+			for (auto var : varset1)
+				if (var->get_type() == TYPED_VARIABLE_LINK)
+					varset2.insert(var->getOutgoingAtom(0));
+		}
+
+		for (Handle clause : clauses)
+			if (is_constant(varset2,clause))
+				if (ref_as.get_atom(clause) == Handle::UNDEFINED)
+					return HandleSet(results.begin(), results.end());
```

Comment:
Since `results` is an `HandleSet`, the returned type, you don't need to create a construct a new `HandleSet`, you can just return it, although, I can see that this what the last line is doing. ;-) I can only assume that the return type was changed, and the last line not properly updated.

Reply:
Yea i just saw it being done like that in the last line so i just did it the same way. Should have added a note to ask you about it cause i did find it strange.